src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -102,22 +106,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   }
 
   private String getLink(String id, String query) {
-    String cse_request = CSE_URL + "?key=" + API_KEY + "&cx=" + id + "&q=" + query;
-    HttpRequest linkRequest =
-        HttpRequest.newBuilder()
-            .GET()
-            .uri(URI.create(cse_request))
-            .setHeader("User-Agent", "Java 11 HttpClient Bot")
-            .build();
-    HttpResponse<String> linkResponse;
-    try {
-      linkResponse = httpClient.send(linkRequest, HttpResponse.BodyHandlers.ofString());
-    } catch (IOException | InterruptedException ex) {
+    String url = CSE_URL + "?key=" + API_KEY + "&cx=" + id + "&q=" + query;
+    try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+      try (CloseableHttpResponse response = httpClient.execute(new HttpGet(url))) {
+
+        if (response.getStatusLine().getStatusCode() != 200) {
+          return null;
+        }
+        HttpEntity entity = response.getEntity();
+        if (entity == null) {
+          return null;
+        }
+        JsonParser parser = new JsonParser();
+        JsonObject obj;
+        try {
+          obj = parser.parse(new InputStreamReader(entity.getContent())).getAsJsonObject();
+        } catch (JsonIOException
+            | JsonSyntaxException
+            | UnsupportedOperationException
+            | IOException e) {
+          return null;
+        }
+        if (!obj.isJsonNull()
+            && !obj.get(CSE_ITEMS).getAsJsonArray().isJsonNull()
+            && obj.get(CSE_ITEMS).getAsJsonArray().size() > 0) {
+          JsonObject item = obj.get(CSE_ITEMS).getAsJsonArray().get(0).getAsJsonObject();
```

Comment:
Ditto here. Though it'd be nice if you could have a more descriptive variable name than `item`.

Reply:
It gets the "item" field of the JSON.